GoogleForCreators/web-stories-wp
# Dashboard: Display dates relative to proper timezone

## Summary
The dashboard was running into an issue where the browser's timezone was interfering with the wordPress set timezone to occasionally show that stories were modified in the future which shouldn't be a thing.

## Relevant Technical Choices
- Remove dashboard specific date formatting util to use the higher level date formatting from `assets/src/dates`.
- Swap dates getting used in dashboard stories serializer to use the `_gmt` version to be in line with editor and make sure we have access to timezone info. 
- Use `.parseZone` in Moment formatting to honor the existing timezones passed in and not touch any browser specific timezone manipulation (this was the root of the "future" dates getting formatted).
- renamed the `dateFormat` prop that's passed from config into an object called `dateSettings` to also pass other date related configs - timezone, gmtOffset, and timeFormat. timeFormat isn't relevant yet but it will be at some point so this is just doing that work so it's in the right spot when it's needed. 
- moment-timezone has all the same info as moment (plus timezones!) so I updated dashboard imports to use moment-timezone to eventually eliminate the excess dependency on moment. 

## User-facing changes
- modified, published, and scheduled stories should now never have future times and the dates/times should be consistent regardless of timezone setting. 

## Testing Instructions
- make a new story, edit a story, publish a story in one timezone and then switch timezones (wp-admin/options-general.php) and see that the relative time since modified/published/scheduled is the same. 
- Also the date formatting should now be consistent. You should be able to change your date format (again in wordPress settings) and have the date format reflect in modified/scheduled/published dates that are not today or yesterday. 

---



Addresses #2199
